src/runtime/pay-client.js

```diff
@@ -222,7 +223,14 @@ export class PayClient {
       )
       .catch(reason => {
         if (typeof reason == 'object' && reason['statusCode'] == 'CANCELED') {
-          return Promise.reject(createCancelError(this.win_));
+          const error = createCancelError(this.win_);
+          if (request) {
+            error['productType'] =
+              /** @type {!PaymentDataRequest} */ (request)['i']['productType'];
+          } else {
+            error['productType'] = ProductType.SUBSCRIPTION;
```

Comment:
Would prefer not to default to anything.

Reply:
Not defaulting to anything makes the test a bit trickier to write.. Would it be okay for me to set it to null? Either way, pay-flow.js ends up defaulting it to the subscription flow if it's not the contribution flow.

Reply:
I've set it to be null by default if no product type is given.